### PR TITLE
Rails5.2

### LINKED ISSUE #82: rails db:setup doesn't mark data migrations completed using data_schema.rb
I've been running into an issue with `rails db:setup` command not loading the `data_schema.rb` file, like it does for `schema.rb`. Basically, when dropping my development database and setting it up again, `rails data:migrate:status` shows all previous migrations as incomplete, even though the `data_schema.rb` file contains the last version number.

I also tried running the tasks by _unpacking setup_: `db:create`, and `db:schema:load:with_data` in order, with no change.

I've gone through the code a bit, and noticed that the `data:schema:load:with_data` task runs `DataMigrate::DatabaseTasks.load_schema_current`, which is an ActiveRecord method. However, this method internally uses another method `schema_file(format)` for determining the file which should be used to load the version.

However,  `DataMigrate::DatabaseTasks` only defined a `self.data_schema_file()` method which, as far as I can tell, is never used except for dumping the version after a data migration is executed. Shouldn't this class override ActiveRecord's `schema_file(format)` method instead of defining a new method?

I tried a simple method rename to override the original implementation, but that didn't have any effect. Do you have any idea what could be going wrong?

### LINKED ISSUE #83: Rails 5.2.0.rc1 breaks data migrations
Just a heads up, not a fix (unfortunately): Migrations have been refactored to better support multiple databases which breaks all data migration tasks.

https://github.com/rails/rails/commit/a2827ec9811b5012e8e366011fd44c8eb53fc714

Result of running the `data:migrate` task:
```
NoMethodError: undefined method 'migrate' for DataMigrate::DataMigrator:Class
.../gems/data_migrate-3.4.0/tasks/databases.rake:249:in 'block (2 levels) in <main>'
```

rails: 5.2.0.rc1
data_migrate: 3.4.0

### PR DESCRIPTION
**Pull request by ilyakatz**
Fixes #83 

| Task        | 5.2           | 4.2  | Notes |
| ------------- |:-------------:| -----:|-----|
|rake data:dump      |  ✅ | ✅ |
|rake data:forward      |  ✅   | ✅ |  |
|rake data:migrate | ✅  |✅|
|rake data:migrate:down | ✅  |✅|  |
|rake data:migrate:up | ✅  |✅|  |
|rake data:migrate:redo | ✅  |✅|  |
|rake data:migrate:status | ✅  |✅|
|rake data:rollback |✅|✅|  |
|rake data:version  | ✅ |✅| 
|rake db:forward:with_data   | ✅  |✅|    |
|rake db:migrate:down:with_data| ✅|✅| |
|rake db:migrate:redo:with_data| ✅|✅||
|rake db:migrate:status:with_data | ✅|✅||
|rake db:migrate:up:with_data| ✅|✅||
|rake db:migrate:with_data| ✅|✅|
|rake db:rollback:with_data | ✅|✅| 
|rake db:schema:load:with_data| ✅  |✅ ||
|rake db:version:with_data | ✅|✅|

### CODE REVIEW

File: lib/data_migrate/migration_context.rb [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+      else
+        []
+      end
+    end
+
+    def move(direction, steps)
+      migrator = DataMigrator.new(direction, migrations)
+
+      if current_version != 0 && !migrator.current_migration
+        raise UnknownMigrationVersionError.new(current_version)
```
Comment (chagar) [inline lib/data_migrate/migration_context.rb]: This should be ActiveRecord::UnknownMigrationVersionError

File: lib/data_migrate/migration_context.rb [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+      paths = Array(migrations_paths)
+      Dir[*paths.flat_map { |path| "#{path}/**/[0-9]*_*.rb" }]
+    end
+
+    def migrations_status
+      db_list = DataSchemaMigration.normalized_versions
+
+      file_list = migration_files.map do |file|
+        version, name, scope = parse_migration_filename(file)
+        raise IllegalMigrationNameError.new(file) unless version
```
Comment (chagar) [inline lib/data_migrate/migration_context.rb]: This should be ActiveRecord::IllegalMigrationNameError

File: lib/data_migrate.rb [hunk 3]
```diff
@@ -1,15 +1,29 @@
 # frozen_string_literal: true
 
-require File.join(File.dirname(__FILE__), "data_migrate", "data_migrator")
+require "pry"
```
Comment (zakmysta) [inline lib/data_migrate.rb]: Is this `require` needed?

File: tasks/databases.rake [hunk 4]
```diff
@@ -241,13 +243,11 @@ namespace :db do
   end
 end
 
+require 'pry'
```
Comment (zakmysta) [inline tasks/databases.rake]: I believe this one isn't needed either.

File: lib/data_migrate.rb (hunk 3 above)
Comment (ilyakatz) [reply lib/data_migrate.rb]: oh yeah, good point! i'm still working on this branch, so need pry to debug, i'll remove them before deploying to rubygems. this branch is not production ready yet!

**Comment by ilyakatz** [comment]
@zakmysta i removed pry require, so if you want to play with the branch, feel free. 

I made some progress with the branch but still a bunch of rake tasks are not working. more work to be done

File: lib/generators/data_migration/data_migration_generator.rb [hunk 5]
```diff
@@ -28,6 +28,8 @@ def set_local_assigns!
       def migration_base_class_name
         if ActiveRecord.version >= Gem::Version.new("5.0")
           "ActiveRecord::Migration[#{ActiveRecord::Migration.current_version}]"
+        elsif ActiveRecord.version >= Gem::Version.new("5.2")
+          "DataMigrate::MigrationContext"
```
Comment (quainjn) [inline lib/generators/data_migration/data_migration_generator.rb]: Running Rails 5.2 and everything is working properly with the generators. However, doesn't seem like this code path could actually get used.

**Pull request merged**